noxfile.py

```diff
@@ -19,7 +26,13 @@
 nox.options.stop_on_first_error = True
 nox.options.sessions = ["fmt", "lint", "check", "test"]
 
-REQUIRES_PYTHON_VERSION = "3.12"
+# N.B.: When updating these, update the corresponding values for the PythonBuildStandalone provider
+# in lift.toml.
+PBS_RELEASE = "20240814"
+PBS_VERSION = "3.12.5"
+PBS_FLAVOR = "install_only_stripped"
```

Comment:
Could this read them directly from `lift.toml`?

Reply:
Oh boy did you open a can of worms there!

I had left this in a half-thought-through state. You can't. You can't even import `tomllib` like I did. There is no guaranty the interpreter that parses the `noxfile.py` is Python 3.12 (or even 3.11) - it will be `nox`'s interpreter which could be `Python>=3.7` today.

I've added a hard limit of 3.8 to avoid extra complexity; then re-worked this and tested where `nox` is hosted by a Python 3.8 x86-64 (extra possible twist) interpreter on Windows ARM64. All is good now. Any developer on any of the science supported platforms with nox on their PATH hosted by Python >=3.8 should be able to hack on the project cleanly... if they also have a Python 3.12 installed somewhere on the PATH.

I think I'll circle back to drop that requirement and have all executions always download a PBS matching the prod version and use that just like Windows ARM64 is doing uniquely right now. This PR is big enough though I think right now.

Thanks @huonw.

Reply:
Ok, the circle-back to always dogfood the prod PBS interpreter for all nox sessions is in #87.